File: app/src/main/java/com/llj/architecturedemo/db/model/MobileEntity.java
package com.llj.architecturedemo.db.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/6/6
 */
@Entity(
        tableName = "mobile",
        indices = @Index("id")
)
public class MobileEntity {
    @PrimaryKey
    private long id;
    private String phone;
    private String province;
    private String city;
    private String service_provider;
    private String city_code;
    private String postcode;
    private int    ret;
    private String searchStr;
    private String operator;
    private String from;
    private String ip;
    private String ua;

    public MobileEntity(long id) {
        this.id = id;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getService_provider() {
        return service_provider;
    }

    public void setService_provider(String service_provider) {
        this.service_provider = service_provider;
    }

    public String getCity_code() {
        return city_code;
    }

    public void setCity_code(String city_code) {
        this.city_code = city_code;
    }

    public String getPostcode() {
        return postcode;
    }

    public void setPostcode(String postcode) {
        this.postcode = postcode;
    }

    public int getRet() {
        return ret;
    }

    public void setRet(int ret) {
        this.ret = ret;
    }

    public String getSearchStr() {
        return searchStr;
    }

    public void setSearchStr(String searchStr) {
        this.searchStr = searchStr;
    }

    public String getOperator() {
        return operator;
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public String getIp() {
        return ip;
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public String getUa() {
        return ua;
    }

    public void setUa(String ua) {
        this.ua = ua;
    }
}

File: app/src/p_app/main/java/com/llj/architecturedemo/db/entity/MobileEntity.kt
package com.llj.architecturedemo.db.entity

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/6/6
 */
@Entity(tableName = "mobile",
        indices = [Index("id")])
data class MobileEntity constructor(@field:PrimaryKey
                                    var id: Long) {
    var phone: String? = null
    var province: String? = null
    var city: String? = null
    @field:SerializedName("service_provider")
    var serviceProvider: String? = null
    @field:SerializedName("city_code")
    var cityCode: String? = null
    var postcode: String? = null
    var ret: Int = 0
    var searchStr: String? = null
    var operator: String? = null
    var from: String? = null
    var ip: String? = null
    var ua: String? = null

}
